auction-app-backend/src/main/java/com/atlantbh/auctionappbackend/security/SecurityConfig.java

```diff
@@ -0,0 +1,70 @@
+package com.atlantbh.auctionappbackend.security;
+
+import com.atlantbh.auctionappbackend.api.AuthWhitelistConfig;
+import com.atlantbh.auctionappbackend.filter.CustomAuthenticationFilter;
+import com.atlantbh.auctionappbackend.filter.CustomAuthorizationFilter;
+import com.auth0.jwt.algorithms.Algorithm;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter;
+import org.springframework.security.web.csrf.CookieCsrfTokenRepository;
+
+import static org.springframework.security.config.http.SessionCreationPolicy.STATELESS;
+
+@Configuration
+@EnableWebSecurity
+public class SecurityConfig extends WebSecurityConfigurerAdapter {
+    private final UserDetailsService userDetailsService;
+    private final PasswordEncoder passwordEncoder;
+    private final JwtConfig jwtConfig;
+    private final Algorithm signAlgorithm;
+    private final String apiPrefix;
+
+    @Autowired
+    public SecurityConfig(UserDetailsService userDetailsService,
+                          PasswordEncoder passwordEncoder,
+                          JwtConfig jwtConfig, Algorithm signAlgorithm,
+                          String apiPrefix, String[] authWhitelist) {
```

Comment:
Formatting (in case of the methods with parameters which do not fit in one line):
```
 public SecurityConfig(
                            UserDetailsService userDetailsService,
                            PasswordEncoder passwordEncoder,
                            JwtConfig jwtConfig, 
                            Algorithm signAlgorithm,
                            String apiPrefix, 
                            String[] authWhitelist
)
```

Reply:
OK I'll keep that in mind.